mmv1/third_party/terraform/utils/config.go.erb

```diff
@@ -128,10 +128,51 @@ type Config struct {
 	GkeHubBasePath string
 }
 
+<% products.each do |product| -%>
+const <%= product[:definitions].name -%>BasePathKey = "<%= product[:definitions].name -%>"
+<% end -%>
+const CloudBillingBasePathKey = "CloudBilling"
+const ComposerBasePathKey = "Composer"
+const ComputeBetaBasePathKey = "ComputeBeta"
+const ContainerBasePathKey = "Container"
+const DataprocBetaBasePathKey = "DataprocBeta"
+const ContainerBetaBasePathKey = "ContainerBeta"
+const DataflowBasePathKey = "Dataflow"
+const IAMBasePathKey = "IAM"
+const IamCredentialsBasePathKey = "IamCredentials"
+const ResourceManagerV2BasePathKey = "ResourceManagerV2"
+const ServiceNetworkingBasePathKey = "ServiceNetworking"
+const StorageTransferBasePathKey = "StorageTransfer"
+const BigtableAdminBasePathKey = "BigtableAdmin"
+const EventarcBasePathKey = "Eventarc"
+const GkeHubFeatureBasePathKey = "GkeHubFeature"
+
+
 // Generated product base paths
+var DefaultBasePaths = map[string]string{
 <% products.each do |product| -%>
-var <%= product[:definitions].name -%>DefaultBasePath = "<%= product[:definitions].base_url -%>"
+	<%= product[:definitions].name -%>BasePathKey : "<%= product[:definitions].base_url -%>",
+<% end -%>
+	CloudBillingBasePathKey : "https://cloudbilling.googleapis.com/v1/",
+<% if version == "ga" -%>
+	ComposerBasePathKey : "https://composer.googleapis.com/v1/",
+<% else -%>
+	ComposerBasePathKey : "https://composer.googleapis.com/v1beta1/",
 <% end -%>
+	ComputeBetaBasePathKey : "https://www.googleapis.com/compute/beta/",
+	ContainerBasePathKey : "https://container.googleapis.com/v1/",
+	ContainerBetaBasePathKey : "https://container.googleapis.com/v1beta1/",
+	DataprocBetaBasePathKey : "https://dataproc.googleapis.com/v1beta2/",
+	DataflowBasePathKey : "https://dataflow.googleapis.com/v1b3/",
+	IAMBasePathKey : "https://iam.googleapis.com/v1/",
+	IamCredentialsBasePathKey : "https://iamcredentials.googleapis.com/v1/",
+	ResourceManagerV2BasePathKey : "https://cloudresourcemanager.googleapis.com/v2/",
+	ServiceNetworkingBasePathKey : "https://servicenetworking.googleapis.com/v1/",
+	StorageTransferBasePathKey : "https://storagetransfer.googleapis.com/v1/",
+	BigtableAdminBasePathKey : "https://bigtableadmin.googleapis.com/v2/",
+	EventarcBasePathKey : "https://eventarc.googleapis.com/v1beta1/",
```

Comment:
This should switch based on the version we are compiling at

Reply:
this key isn't used... dcl actually internally hardcodes their own basepath. I remove this reference !